### PR TITLE
Fix generation of null maps so that we get a good db file

### PR DESCRIPTION
**Pull request by jcpunk**
Alternate solution for patch #135

### CODE REVIEW

**Comment by igalic** [comment]
we're still missing a couple of fixes in the spec tests: https://travis-ci.org/camptocamp/puppet-postfix/jobs/139011500#L1340

but the acceptance tests are failing because… docker.

**Comment by jcpunk** [comment]
I've got the tests that currently run fixed.

File: manifests/hash.pp [hunk 1]
```diff
@@ -66,7 +66,8 @@
 
   file {"${name}.db":
     ensure  => $ensure,
-    require => [File[$name], Exec["generate ${name}.db"]],
+    require => File[$name],
+    notify  => Exec["generate ${name}.db"],
```
Comment (raphink) [inline manifests/hash.pp]: So this actually changes the ordering of the Exec and File. Is that right?

**Comment by raphink** [comment]
@igalic apparently, to the exception of some Debian/Ubuntu versions that fail to start the service idempotently, the errors in acceptance tests come from serverspec.

File: manifests/hash.pp (hunk 1 above)
Comment (jcpunk) [inline manifests/hash.pp]: Correct, otherwise puppet will create a zero byte file for us which is the opposite of what we want.

**Pull request closed without merging**